datasets/kz/kz_companies/crawler.py

```diff
@@ -0,0 +1,120 @@
+from time import sleep
+
+from zavod import Context
+from zavod import helpers as h
+from rigour.urls import build_url
+
+
+BASE_URL = "https://data.egov.kz/datasets/view?index=gbd_ul"
+CRAWL_URL = "https://data.egov.kz/datasets/getdata"
+CACHE_DAYS = 30
+SLEEP_TIME = 1
+
+FORMDATA = {
+    "index": "gbd_ul",
+    "version": "v1",
+    "page": "1",
+    "count": "20",
+    "text": "",
+    "column": "id",
+    "order": "ascending",
+}
+
+FIELDS_MAPPING = {
+    "datereg": "incorporationDate",
+    "nameru": "name",
+    "addresskz": "address",
+    "namekz": "name",
+    "bin": "registrationNumber",
+    "addressru": "address",
+    "statuskz": "status",
+    "statusru": "status",
+    "okedru": "notes",
+    "okedkz": "notes",
+}
+
+
+HEADERS = {
+    "Referer": BASE_URL,
+    "X-Requested-With": "XMLHttpRequest",
+}
+
+
+def crawl_page(context: Context, page_number: int) -> int:
+    """
+    Crawls a single page of the Kazakhstan data portal and emits the data.
+    Args:
+        context: The context object for the current dataset.
+        page_number: The page number to crawl.
+    Returns:
+        The total number of pages.
+    """
+
+    formdata = FORMDATA.copy()
+    formdata["page"] = str(page_number)
+
+    data = context.fetch_json(
+        CRAWL_URL,
+        cache_days=CACHE_DAYS,
+        headers=HEADERS,
+        params=formdata,
+    )
+
+    for item in data["elements"]:
+        if item["datereg"]:
+            plus_idx = item["datereg"].find("+")
+            item["datereg"] = h.parse_date(
+                item["datereg"][0:plus_idx] if plus_idx != -1 else item["datereg"],
+                ["%Y-%m-%d"],
+            )[0]
+
+        entity = context.make("Company")
+        entity.id = context.make_id("KZCompany", item["id"], item["bin"])
+
+        for k, v in FIELDS_MAPPING.items():
+            if k in item and item.get(k):
+                lang = None
+                if k.endswith("ru"):
+                    lang = "rus"
+                elif k.endswith("kz"):
+                    lang = "kaz"
+
+                entity.add(v, item.get(k), lang=lang)
+
+        entity.add("country", "kz")
+        entity.add("sourceUrl", build_url(CRAWL_URL, params=formdata))
```

Comment:
it doesn't look like these URLs work with a plain GET, so I think it's more likely to be a bit frustrating for users than helpful. Perhaps best to skip. They'd have to come check the source to see the creative way you figured out to get a useful response :)

Reply:
Ok, will remove it.